### PR TITLE
Cron job for : Add missing products to the index

### LINKED ISSUE #29085: Cron job for : Add missing products to the index
### Prerequisites

- [X] I understand and accept the project's [code of conduct](https://github.com/PrestaShop/PrestaShop/blob/develop/CODE_OF_CONDUCT.md)
- [X] I have already [searched in existing features request](https://github.com/PrestaShop/PrestaShop/issues?q=is%3Aissue+label%3AImprovement%2CFeature%2CDeveloper+Feature+) and found no previous suggestion of this feature.

### Is your feature request related to a problem?

Would it be possible to add a cron job for updating missing products to index? ( search page )

I am aware the we have this for rebuild the full index, but for stores that have large amount of products, this option sometime stops working ( in my case ) and leaves the index incomplete. 

### Describe the solution you'd like

a cron job for updating missing products to index

### Alternatives you've considered

n/a

### Additional context

_No response_

### Do you plan to work on this feature?

- [X] I'm willing to contribute a [formal specification](https://github.com/PrestaShop/prestashop-specs).
- [X] I'm willing to provide any wireframes or design assets required for this feature.
- [X] I'm willing to submit a Pull Request that implements this feature.
- [X] I'm willing to help verify that the implemented feature works as intended and produces no unintended side effects.

### PR DESCRIPTION
**Pull request by ghost**
| Questions         | Answers
| ----------------- | -------------------------------------------------------
| Branch?           | develop
| Description?      | A cron job for updating missing products to index
| Type?             | improvement
| Category?         | BO
| BC breaks?        | no
| Deprecations?     | no
| Fixed ticket?     | Fixes #29085.
| Related PRs       | 
| How to test?      | Go to `Shop Parameters` -> `Search` and check if a new url for missing product
| Possible impacts? |

### CODE REVIEW

File: controllers/admin/AdminSearchConfController.php [hunk 1]
```diff
@@ -106,11 +115,18 @@ public function __construct()
...
+						<a href="' . Tools::safeOutput($cronUrlFullIndex) . '">
 							<i class="icon-external-link-sign"></i>
-							' . Tools::safeOutput($cron_url) . '
+							' . Tools::safeOutput($cronUrlFullIndex) . '
 						</a>
-					</p><br />',
+					</p>
+                    <p>
+                        ' . $this->trans('You can set a cron job that will index your missing products using the following URL:', [], 'Admin.Shopparameters.Feature') . '<br />
+                        <a href="' . Tools::safeOutput($conUrlMissingProduct) . '">
```
Comment (PrestaEdit) [inline controllers/admin/AdminSearchConfController.php]: Missing a r. (con -> cron)

File: controllers/admin/AdminSearchConfController.php [hunk 2]
```diff
@@ -75,7 +75,16 @@ public function __construct()
...
+        $cronUrlFullIndex = Context::getContext()->link->getAdminLink(
+            'AdminSearch',
+            false,
+            [],
+            $params
+        );
+
+        unset($params['full']);
+
+        $conUrlMissingProduct = Context::getContext()->link->getAdminLink(
```
Comment (PrestaEdit) [inline controllers/admin/AdminSearchConfController.php]: Missing a r. (con -> cron)

File: controllers/admin/AdminSearchConfController.php [hunk 3]
```diff
@@ -106,11 +115,18 @@ public function __construct()
...
-							' . Tools::safeOutput($cron_url) . '
+							' . Tools::safeOutput($cronUrlFullIndex) . '
 						</a>
-					</p><br />',
+					</p>
+                    <p>
+                        ' . $this->trans('You can set a cron job that will index your missing products using the following URL:', [], 'Admin.Shopparameters.Feature') . '<br />
+                        <a href="' . Tools::safeOutput($conUrlMissingProduct) . '">
+                            <i class="icon-external-link-sign"></i>
+                            ' . Tools::safeOutput($conUrlMissingProduct) . '
```
Comment (PrestaEdit) [inline controllers/admin/AdminSearchConfController.php]: Missing a r. (con -> cron)

**Review by Hlavtox — APPROVED** [review]

**Comment by 01park** [comment]
Hi,

I have fixed the improved file, however I can see this error in my log.

Is it related?

SearchCore::indexation($full = '1', $id_product = *uninitialized*)/home/*******/public_html/controllers/admin/AdminSearchController.php:522

**Review by kpodemski — APPROVED** [review]

File: controllers/admin/AdminSearchConfController.php [hunk 4]
```diff
@@ -106,11 +115,18 @@ public function __construct()
...
-						<a href="' . Tools::safeOutput($cron_url) . '">
+						<a href="' . Tools::safeOutput($cronUrlFullIndex) . '">
 							<i class="icon-external-link-sign"></i>
-							' . Tools::safeOutput($cron_url) . '
+							' . Tools::safeOutput($cronUrlFullIndex) . '
 						</a>
-					</p><br />',
+					</p>
+                    <p>
+                        ' . $this->trans('You can set a cron job that will index your missing products using the following URL:', [], 'Admin.Shopparameters.Feature') . '<br />
```
Comment (kpodemski) [inline controllers/admin/AdminSearchConfController.php:124]: Could you check the wording @l-delin ? It is a simple differentiation between a full indexation and indexing only the missing products :)

Comment (Julievrz) [reply controllers/admin/AdminSearchConfController.php:124]: image
Hi @kpodemski. There's something I don't understand: doesn't this option already exist?

Comment (kpodemski) [reply controllers/admin/AdminSearchConfController.php:124]: The one that exists is about reindexing everything, and this PR is about adding the second option to index what's missing only

**Pull request closed without merging**